### DISCUSSION TITLE
Nethermind -

### DISCUSSION THREAD

**Discussion by Crypto6Almonds** [zkSync-Community-Hub/zksync-developers#342] [Dev Tooling] [answered]
When using the Nethermind plugin with remix does the contract get deployed to the Era chain and if so why does it not trigger a separate Metamask transaction? 

I've clicked deploy and MM ask me to sign permissions and it appears in Remix that the contract is live, but I cannot tell because only the subsequent actions are visible in the explorer. TY

**Comment by jaguard2021** [comment]
Most likely it is from the structure of your programming code. Usually there is a certain part of the code in your program that is not compatible with the latest metamask version.
One example is that using web3 ver 1.2.2 (typescript's @types/web3) is not compatible with the latest version of metamask.

**Comment by dutterbutter** [comment, marked as answer]
@Crypto6Almonds the remix plugin uses a specific devnet, and not zkSync Era so you would not see the deployment on the explorer.

**Comment by lysumeduso** [comment]
When using the Nethermind plugin with Remix, the contract deployment does occur on the zkSync (Era) chain. However, since zkSync transactions are batched and processed off-chain, they don't individually trigger separate MetaMask transactions. Instead, you'll see the results of your contract deployment and subsequent actions reflected in the zkSync Explorer.

**Comment by phicurby** [comment]
When deploying a contract using the Nethermind plugin with Remix, the contract is deployed to the Era chain. However, it may not trigger a separate Metamask transaction because the plugin interacts directly with the blockchain node (Nethermind), bypassing the need for Metamask to initiate transactions. Therefore, while the contract deployment appears successful in Remix, it may not be reflected as a separate transaction in Metamask.

**Comment by shipefred** [comment]
When using the Nethermind plugin with Remix, the contract is deployed to the Era chain. However, it may not trigger a separate Metamask transaction because the Nethermind plugin may handle contract deployment differently, bypassing the need for a separate Metamask transaction. As a result, while the contract deployment is successful and visible in Remix, it may not be directly reflected as a separate transaction in Metamask.

**Comment by davehutchinsv** [comment]
The user is asking if a contract deployed using the Nethermind plugin with Remix gets deployed to the Era chain and why it does not trigger a separate Metamask transaction. After clicking deploy and signing permissions with Metamask, the user sees that the contract is live in Remix, but subsequent actions are only visible in the explorer. Further investigation and debugging may be needed to determine the cause of the issue and the feasibility and impact of any necessary changes.
